src/crypto/algorithms/megolm.js

```diff
@@ -543,6 +547,9 @@ MegolmDecryption.prototype.decryptEvent = function(event) {
     } catch (e) {
         if (e.message === 'OLM.UNKNOWN_MESSAGE_INDEX') {
             this._addEventToPendingList(event);
+            if (requestKeysOnFail) {
+                this._requestKeysForEvent(event);
+            }
```

Comment:
well, it's fine for now. Until the key turns up it is just an undecryptable event.

decryptEvent is called [here](https://github.com/matrix-org/matrix-js-sdk/blob/00a46273bc0add9a7088ca8a0a0c12deec01b345/src/crypto/index.js#L783) which in turn called [here](https://github.com/matrix-org/matrix-js-sdk/blob/00a46273bc0add9a7088ca8a0a0c12deec01b345/src/client.js#L601).

I was toying with setting a flag on the event to show that its key had been requested or something, but I didn't need it yet.